### PR TITLE
HPCC-16296 Add "undef" linting

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq Please Review

File: esp/src/eclwatch/DynamicESDLMethodWidget.js [hunk 1]
```diff
@@ -211,7 +211,7 @@ define([
             }).then(function (response) {
                 if (lang.exists("ConfigureESDLBindingMethodResponse.status", response)) {
                     if (response.ConfigureESDLBindingMethodResponse.status.Code === 0) {
-                        context.store.remove(selection[idx].__hpcc_id);
+                        //  MV Is this needed (idx is invalid)?:  context.store.remove(selection[idx].__hpcc_id);
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLMethodWidget.js]: @miguelvazq idx is invalid here and given you call response below, is the store.remove code neede?

**Review by GordonSmith — COMMENTED** [review]
One question for @miguelvazq

**Comment by GordonSmith** [comment]
@miguelvazq bump - see question above.

**Comment by GordonSmith** [comment]
@miguelvazq  bump - see question above.

File: esp/src/eclwatch/DynamicESDLMethodWidget.js (hunk 1 above)
Comment (miguelvazq) [inline esp/src/eclwatch/DynamicESDLMethodWidget.js]: Sorry been working on a few things, I have not forgotten about this. Let me test with some DESDL examples and respond here.

Comment (miguelvazq) [inline esp/src/eclwatch/DynamicESDLMethodWidget.js]: No @GordonSmith that is idx is not needed.

Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLMethodWidget.js]: @miguelvazq Can you open a Jira and fix this source?

**Comment by richardkchapman** [comment]
Needs rebasing for a clean merge

**Comment by richardkchapman** [comment]
@GordonSmith Please rebase.

**Comment by GordonSmith** [comment]
Rebased - @miguelvazq please re-review.

File: esp/src/eclwatch/DynamicESDLMethodWidget.js (hunk 1 above)
Comment (miguelvazq) [inline esp/src/eclwatch/DynamicESDLMethodWidget.js]: It will be rectified in HPCC-16613. As users will input the xml they want to create/remove.

**Comment by miguelvazq** [comment]
I see there is missing `var` being added and `===` being swapped is there a reason why we are adding missing portions of code?

**Comment by GordonSmith** [comment]
@miguelvazq They were fixing newly introduced linting issues from other branches.

IOW This branch currently has no linting issues including the newly introduced "undef" lint test.  
In the future, before we issue a new PR, we should try and run ```npm run test``` locally and it will report any potential issues.  (you may need to install node and run ```npm install`` first)

File: esp/src/eclwatch/WsWorkunits.js [hunk 2]
```diff
@@ -141,11 +141,6 @@ define([
             return ESPRequest.send("WsWorkunits", "WUGetNumFileToCopy", params);
         },
 
-        CreateQuerySetStore: function (options) {
```
Comment (miguelvazq) [inline esp/src/eclwatch/WsWorkunits.js]: What happens to anything calling this? Or does the linting look for calls that are unused?

Comment (GordonSmith) [inline esp/src/eclwatch/WsWorkunits.js]: In this case the linting highlighted that the content of this function was invalid and I searched and no-one is calling it.

**Comment by GordonSmith** [comment]
@miguelvazq bump.

**Comment by miguelvazq** [comment]
Looks fine.

**Pull request merged**